### ISSUE TITLE
Test failure on PackageEvaluator, julia nightly

### ISSUE THREAD

**Issue by tkelman** [BioJulia/BufferedStreams.jl#20]
```
>>> 'Pkg.add("BufferedStreams")' log
INFO: Installing BufferedStreams v0.1.4
INFO: Package database updated
INFO: METADATA is out-of-date — you may not have the latest version of BufferedStreams
INFO: Use `Pkg.update()` to get the latest versions of your packages

>>> 'Pkg.test("BufferedStreams")' log
Julia Version 0.4.6
Commit 2e358ce (2016-06-19 17:16 UTC)
Platform Info:
  System: Linux (x86_64-unknown-linux-gnu)
  CPU: Intel(R) Xeon(R) CPU E3-1241 v3 @ 3.50GHz
  WORD_SIZE: 64
  BLAS: libopenblas (USE64BITINT DYNAMIC_ARCH NO_AFFINITY Nehalem)
  LAPACK: libopenblas64_
  LIBM: libopenlibm
  LLVM: libLLVM-3.3
INFO: Computing test dependencies for BufferedStreams...
INFO: Installing BaseTestNext v0.2.1
INFO: Testing BufferedStreams
seekforward: Error During Test
  Test threw an exception of type ArgumentError
  Expression: all(Bool[test_seekforward(stream,position,offset) for (position,offset) = zip(positions,offsets)])
  ArgumentError: n must be non-negative in seekforward
   in test_seekforward at /home/vagrant/.julia/v0.4/BufferedStreams/test/runtests.jl:239
   in anonymous at /home/vagrant/.julia/v0.4/BaseTestNext/src/BaseTestNext.jl:165
   in do_test at /home/vagrant/.julia/v0.4/BaseTestNext/src/BaseTestNext.jl:181
   [inlined code] from /home/vagrant/.julia/v0.4/BufferedStreams/test/runtests.jl:248
   in anonymous at no file:0
   in include at ./boot.jl:261
   in include_from_node1 at ./loading.jl:320
   in process_options at ./client.jl:280
   in _start at ./client.jl:378
Test Summary:       | Pass  Error  Total
BufferedInputStream |   67      1     68
  read              |    8             8
  peek              |    4             4
  peekbytes!        |   10            10
  readbytes!        |   10            10
  readuntil         |    2             2
  arrays            |    2             2
  marks             |    7             7
  anchors           |    1             1
  seek              |    2             2
  seekforward       |    2      1      3
  close             |    5             5
  iostream          |   11            11
  misc.             |    3             3
ERROR: LoadError: Some tests did not pass: 67 passed, 0 failed, 1 errored.
 in finish at /home/vagrant/.julia/v0.4/BaseTestNext/src/BaseTestNext.jl:385
 [inlined code] from /home/vagrant/.julia/v0.4/BaseTestNext/src/BaseTestNext.jl:559
 in anonymous at no file:0
 in include at ./boot.jl:261
 in include_from_node1 at ./loading.jl:320
 in process_options at ./client.jl:280
 in _start at ./client.jl:378
while loading /home/vagrant/.julia/v0.4/BufferedStreams/test/runtests.jl, in expression starting on line 18
===========================[ ERROR: BufferedStreams ]===========================

failed process: Process(`/home/vagrant/julia/bin/julia --check-bounds=yes --code-coverage=none --color=no /home/vagrant/.julia/v0.4/BufferedStreams/test/runtests.jl`, ProcessExited(1)) [1]

================================================================================
INFO: Removing BaseTestNext v0.2.1
ERROR: BufferedStreams had test errors
 in error at ./error.jl:21
 in test at pkg/entry.jl:803
 in anonymous at pkg/dir.jl:31
 in cd at file.jl:22
 in cd at pkg/dir.jl:31
 in test at pkg.jl:71
 in process_options at ./client.jl:257
 in _start at ./client.jl:378

>>> End of log
```

Will let you know if it happens often - keep an eye on http://pkg.julialang.org/detail/BufferedStreams.html

**Comment by bicycle1885** [comment]
Just a note: this failure didn't happen in #21 (https://travis-ci.org/BioJulia/BufferedStreams.jl/jobs/141476500). I need to investigate it later.

**Comment by tkelman** [comment]
0.4 issue was probably spurious, but there's a new one on nightly
```
seek: Error During Test
  Got an exception of type UndefVarError outside of a @test
  UndefVarError: position not defined
   in macro expansion; at /home/vagrant/.julia/v0.5/BufferedStreams/test/runtests.jl:213 [inlined]
   in macro expansion; at ./test.jl:672 [inlined]
   in macro expansion; at /home/vagrant/.julia/v0.5/BufferedStreams/test/runtests.jl:208 [inlined]
   in macro expansion; at ./test.jl:672 [inlined]
   in anonymous at ./<missing>:?
   in include_from_node1(::String) at ./loading.jl:426
   in process_options(::Base.JLOptions) at ./client.jl:262
   in _start() at ./client.jl:318
```

**Comment by bicycle1885** [comment]
#22 will fix the problem. I think this is due to a bug of Julia.

**Comment by bicycle1885** [comment]
I think this issue was resolved in #22.
